extide/gradle/src/org/netbeans/modules/gradle/problems/PropertiesEditor.java

```diff
@@ -0,0 +1,141 @@
+import javax.swing.text.Document;
+import org.netbeans.api.actions.Savable;
+import org.netbeans.api.editor.document.AtomicLockDocument;
+import org.netbeans.api.editor.document.LineDocumentUtils;
+import org.openide.cookies.EditorCookie;
+import org.openide.filesystems.FileObject;
+import org.openide.filesystems.FileUtil;
+import org.openide.util.EditableProperties;
+
+/**
+ *
+ * @author sdedic
+ */
+public class PropertiesEditor {
+    private static final Logger LOG = Logger.getLogger(PropertiesEditor.class.getName());
+    
+    private FileObject file;
+    private final Path filePath;
+    private Document openedDocument;
+    private EditableProperties properties;
+
+    public PropertiesEditor(FileObject file) {
+        this.file = file;
+        File f = FileUtil.toFile(file);
+        this.filePath = f != null ? f.toPath() : null;
+    }
+    
+    public PropertiesEditor(Path path) {
+        this.filePath = path;
+    }
+
+    public Path getFilePath() {
+        return filePath;
+    }
+    
+    public EditableProperties open() throws IOException {
+        if (properties != null) {
+            return properties;
+        }
+        if (file == null) {
+            return properties = new EditableProperties(false);
+        }
+        EditorCookie cake = file.getLookup().lookup(EditorCookie.class);
+        if (cake != null) {
+            openedDocument = cake.getDocument();
+        }
+        
+        EditableProperties p;
+        if (openedDocument == null) {
+            try (InputStream istm = file.getInputStream()) {
+                p = new EditableProperties(false);
+                p.load(istm);
+            }
+            this.properties = p;
+            return p;
+        }
+        IOException[] err = new IOException[1];
+        AtomicReference<ByteArrayInputStream> ref = new AtomicReference<>();
+        openedDocument.render(() -> {
+            ByteArrayOutputStream baos = new ByteArrayOutputStream();
+            try (PrintWriter pw = new PrintWriter(baos)) {
+                pw.print(
+                        openedDocument.getText(0, openedDocument.getLength())
+                );
+                ref.set(new ByteArrayInputStream(baos.toByteArray()));
+            } catch (BadLocationException ex) {
+                err[0] = new IOException(ex);
+            }
+        });
+        p = new EditableProperties(false);
+        p.load(ref.get());
+        this.properties = p;
+        return p;
+    }
+    
+    public void save() throws IOException {
+        if (file != null) {
+            EditorCookie cake = file.getLookup().lookup(EditorCookie.class);
+            if (cake != null) {
+                openedDocument = cake.getDocument();
+            }
+
+            if (openedDocument != null) {
+                ByteArrayOutputStream baos = new ByteArrayOutputStream();
+                properties.store(baos);
+                String str = new String(baos.toByteArray(), StandardCharsets.ISO_8859_1);
```

Comment:
Sorry, I don't understand ? Please elaborate on the concern - what changes specifically with JDK17 here ?

@jlahoda pointed out that starting from JDK9, resource bundles accept UTF8-encoded properties by default; but that's for ResourceBundles, which should not generic .properties format which still loads from ISO8859-1

Reply:
Well, I know that from JDK 17 the default JVM encoding is UTF-8. I do not think that ISO_8859_1 would cause any problem, just seemed to be strange. That's all.